src/symbol.rs

```diff
@@ -0,0 +1,89 @@
+use crate::HandleScope;
+use crate::Isolate;
+use crate::Local;
+use crate::String;
+use crate::Symbol;
+use crate::Value;
+
+extern "C" {
+  fn v8__Symbol__New(
+    isolate: *mut Isolate,
+    description: *const String,
+  ) -> *const Symbol;
+  fn v8__Symbol__ForApi(
+    isolate: *mut Isolate,
+    description: *const String,
+  ) -> *const Symbol;
+  fn v8__Symbol__Description(this: *const Symbol) -> *const Value;
+}
+
+macro_rules! well_known {
+  ($name: ident, $binding: ident) => {
+    pub fn $name<'s>(scope: &mut HandleScope<'s>) -> Local<'s, Symbol> {
+      extern "C" {
+        fn $binding(isolate: *mut Isolate) -> *const Symbol;
+      }
+      unsafe { scope.cast_local(|sd| $binding(sd.get_isolate_ptr())) }.unwrap()
+    }
+  };
+}
+
+impl Symbol {
+  /// Create a symbol. If description is not empty, it will be used as the
+  /// description.
+  pub fn new<'s>(
+    scope: &mut HandleScope<'s>,
+    description: Option<Local<String>>,
+  ) -> Local<'s, Symbol> {
+    unsafe {
+      scope.cast_local(|sd| {
+        v8__Symbol__New(
+          sd.get_isolate_ptr(),
+          description.map_or_else(std::ptr::null, |v| &*v),
+        )
+      })
+    }
+    .unwrap()
+  }
+
+  /// Access global symbol registry.
+  /// Note that symbols created this way are never collected, so
+  /// they should only be used for statically fixed properties.
+  /// Also, there is only one global description space for the descriptions used as
+  /// keys.
+  /// To minimize the potential for clashes, use qualified descriptions as keys.
+  pub fn r#for<'s>(
```

Comment:
`r#for` is pretty unfortunate... However let's leave it for now.

Reply:
It's fairly idiomatic to use `for_` to dodge keywords in Rust. I think the `r#` syntax was primarily introduced to make automatic updating of code from Rust 2015 to Rust 2018 easier.

Reply:
I find `for_` also unfortunate. 
Taking the risk of unleashing bikeshed - what about `for_global` ?

Reply:
Here's an [official style guide](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/advice.md#names) mentioning the `for_` style:

> When a name is forbidden because it is a reserved word (e.g., `crate`), use a trailing underscore to make the name legal (e.g., `crate_`)

I've seen it a lot in practice, e.g. [here's a search](https://github.com/rust-lang/rust/search?q=for_&unscoped_q=for_) of all the times `for_` appears in the rustc source itself. Best to stick to the community convention, I think.

Reply:
There are no good options here so I'm cool with whatever.

Reply:
Renamed to `for_global`